modules/study/src/main/NewPgnImport.scala

```diff
@@ -78,52 +86,52 @@ object NewPgnImport:
     }
 
   private def makeTree(
-      setup: chess.Game,
+      context: Context,
       node: ParsedPgnTree,
       annotator: Option[Comment.Author]
   ): Option[PgnNode[NewBranch]] =
-    node.mapAccumlOption_(setup): (setup, data) =>
-      transform(setup, data, annotator)
+    node.mapAccumlOption_(context): (context, data) =>
+      transform(context, data, annotator)
 
   private def transform(
-      context: chess.Game,
+      context: Context,
       data: PgnNodeData,
       annotator: Option[Comment.Author]
-  ): (chess.Game, Option[NewBranch]) =
+  ): (Context, Option[NewBranch]) =
     data
-      .san(context.situation)
+      .san(context.currentGame.situation)
       .map(moveOrDrop =>
-        val game   = moveOrDrop.applyGame(context)
+        val game   = moveOrDrop.applyGame(context.currentGame)
         val uci    = moveOrDrop.toUci
         val id     = UciCharPair(uci)
         val sanStr = moveOrDrop.toSanStr
-        (
-          game,
-          PgnImport.parseComments(data.metas.comments, annotator) match
-            case (shapes, clock, comments) =>
-              NewBranch(
-                id = id,
-                move = Uci.WithSan(uci, sanStr),
-                comp = false,
-                forceVariation = false,
-                Metas(
-                  ply = game.ply,
-                  fen = Fen.write(game),
-                  check = game.situation.check,
-                  dests = None,
-                  drops = None,
-                  eval = None,
-                  shapes = shapes,
-                  comments = comments,
-                  gamebook = None,
-                  glyphs = data.metas.glyphs,
-                  opening = None,
-                  clock = clock,
-                  crazyData = game.situation.board.crazyData
-                )
-              ).some
-        )
+        val newBranch = PgnImport.parseComments(data.metas.comments, annotator) match
+          case (shapes, clock, emt, comments) =>
+            NewBranch(
+              id = id,
+              move = Uci.WithSan(uci, sanStr),
+              comp = false,
+              forceVariation = false,
+              Metas(
+                ply = game.ply,
+                fen = Fen.write(game),
+                check = game.situation.check,
+                dests = None,
+                drops = None,
+                eval = None,
+                shapes = shapes,
+                comments = comments,
+                gamebook = None,
+                glyphs = data.metas.glyphs,
+                opening = None,
+                clock = clock.orElse((context.previousClock, emt).mapN(_ - _)),
```

Comment:
What about clock increments tho? This only works for clocks like x+0.

Or do we think it's better to estimate assuming zero increment, rather than having no clock at all?

Reply:
If PGN sources specified the clock format in the PGN tags, then we could use that, but they rarely do.